src/Controls/src/SourceGen/RouteSourceGenerator.cs

```diff
@@ -0,0 +1,252 @@
+			}
+
+			foreach (var scopedView in routedPages.Where(x => x.Lifetime == Scoped))
+			{
+				foreach (var route in scopedView.Routes)
+				{
+					if (string.IsNullOrEmpty(route))
+					{
+						routes.AppendLine($"{Indent}Routing.RegisterRoute(nameof(global::{GetNamespace(scopedView.Type)}.{scopedView.Type.Identifier}), typeof(global::{GetNamespace(scopedView.Type)}.{scopedView.Type.Identifier}));");
+					}
+					else
+					{
+						routes.AppendLine($"{Indent}Routing.RegisterRoute({route}, typeof(global::{GetNamespace(scopedView.Type)}.{scopedView.Type.Identifier}));");
+					}
+				}
+
+				if (addComment)
+				{
+					addComment = false;
+					services.AppendLine($"{Indent}// Scoped Services");
+				}
+
+				if (scopedView.ImplicitViewModel)
+				{
+					services.AppendLine($"{Indent}builder.Services.AddScoped<{scopedView.Type.Identifier.ValueText.Replace("Page", "ViewModel")}>();");
+				}
+				else if (!string.IsNullOrEmpty(scopedView.ViewModelType))
+				{
+					services.AppendLine($"{Indent}builder.Services.AddScoped<{scopedView.ViewModelType}>();");
+				}
+
+				services.AppendLine($"{Indent}builder.Services.AddScoped<global::{GetNamespace(scopedView.Type)}.{scopedView.Type.Identifier}>();");
+			}
+
+			foreach (var service in syntaxReceiver.Services.Where(x => x.Lifetime == Scoped))
+			{
+				if (string.IsNullOrEmpty(service.RegisterFor))
+				{
+					services.AppendLine($"{Indent}builder.Services.{(service.UseTryAdd ? TryAdd : Add)}Scoped<global::{GetNamespace(service.Type)}.{service.Type.Identifier}>();");
+				}
+				else
+				{
+					services.AppendLine($"{Indent}builder.Services.{(service.UseTryAdd ? TryAdd : Add)}Scoped<{service.RegisterFor}, global::{GetNamespace(service.Type)}.{service.Type.Identifier}>();");
+				}
+			}
+
+			if (!addComment)
+			{
+				addComment = true;
+				services.AppendLine();
+			}
+
+			foreach (var transientView in routedPages.Where(x => x.Lifetime == Transient))
+			{
+				foreach (var route in transientView.Routes)
+				{
+					if (string.IsNullOrEmpty(route))
+					{
+						routes.AppendLine($"{Indent}Routing.RegisterRoute(nameof(global::{GetNamespace(transientView.Type)}.{transientView.Type.Identifier}), typeof(global::{GetNamespace(transientView.Type)}.{transientView.Type.Identifier}));");
+					}
+					else
+					{
+						routes.AppendLine($"{Indent}Routing.RegisterRoute({route}, typeof(global::{GetNamespace(transientView.Type)}.{transientView.Type.Identifier}));");
+					}
+				}
+
+				if (addComment)
+				{
+					addComment = false;
+					services.AppendLine($"{Indent}// Transient Services");
+				}
+
+				if (transientView.ImplicitViewModel)
+				{
+					services.AppendLine($"{Indent}builder.Services.AddTransient<{transientView.Type.Identifier.ValueText.Replace("Page", "ViewModel")}>();");
+				}
+				else if (!string.IsNullOrEmpty(transientView.ViewModelType))
+				{
+					services.AppendLine($"{Indent}builder.Services.AddTransient<{transientView.ViewModelType}>();");
+				}
```

Comment:
No need for the else clause.

Reply:
The same scenario of precedence.

Reply:
Same, refer to the first one

Reply:
Have refactored the code with more switch statements and local functions to avoid nesting and repetition. Kindly review it again and provide your comments.